hw06-atm/src/main/java/ru/otus/atm/ATM.java

```diff
@@ -0,0 +1,84 @@
+package ru.otus.atm;
+
+import lombok.RequiredArgsConstructor;
+import ru.otus.atm.banknote.Banknote;
+import ru.otus.atm.banknote.Ruble;
+import ru.otus.atm.cell.CellRepository;
+import ru.otus.atm.cell.StandardCellRepository;
+import ru.otus.atm.exceptions.InvalidTakeAmountException;
+import ru.otus.atm.exceptions.NegativeAmountException;
+import ru.otus.atm.exceptions.NotEnoughBanknotesException;
+
+import java.util.*;
+import java.util.function.BiFunction;
+import java.util.stream.Collectors;
+
+@RequiredArgsConstructor
+public class ATM {
+    private final CellRepository cellRepository;
+
+    public int getCurrentSum() {
+        return cellRepository.getSum();
+    }
+
+    public Map<Banknote, Integer> takeMoney(int sum) throws InvalidTakeAmountException, NegativeAmountException {
```

Comment:
Правильно, что сделали свои исключения.
И правильно, что `RuntimeException` (от проверяемых исключений все уходят), поэтому можно не указывать `throws`.
```suggestion
    public Map<Banknote, Integer> takeMoney(int sum) {
```

Reply:
throws здесь выполняет роль документации :)

Reply:
Дело ваше, но польза сомнительная.